iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheckTest.java

```diff
@@ -43,30 +46,75 @@ static Stream<String> sensitiveKinds() {
+    var expectedSecondary4 = new SecondaryLocation(range(46, 0, 46, 20), SECONDARY_MESSAGE_VARIABLE);
+    var expectedSecondary5 = new SecondaryLocation(range(48, 0, 48, 20), SECONDARY_MESSAGE_VARIABLE);
     var expectedIssues = List.of(
-      issue(19, 19, 19, 24, PRIMARY_MESSAGE, expectedSecondary1),
-      issue(28, 10, 28, 31, PRIMARY_MESSAGE, expectedSecondary2, expectedSecondary3),
-      issue(50, 16, 50, 20, PRIMARY_MESSAGE, expectedSecondary4, expectedSecondary5));
-    KubernetesVerifier.verify("DuplicatedEnvironmentVariables/DuplicatedEnvsChart/templates/duplicated-env-pod.yaml", check, expectedIssues);
+      issue(19, 19, 19, 24, PRIMARY_MESSAGE_VARIABLE, expectedSecondary1),
+      issue(28, 10, 28, 31, PRIMARY_MESSAGE_VARIABLE, expectedSecondary2, expectedSecondary3),
+      issue(50, 16, 50, 20, PRIMARY_MESSAGE_VARIABLE, expectedSecondary4, expectedSecondary5));
+    KubernetesVerifier.verify("DuplicatedEnvironmentVariables/SingleFileTests/DuplicatedEnvsChart/templates/duplicated-env-pod.yaml", check, expectedIssues);
+  }
+
+  @ParameterizedTest
+  @CsvSource(delimiter = ';', value = {
+    "RepetitionAcrossAllSources;repetition_across_all_sources.yaml;my-config-map.yaml,my-secret.yaml",
+    "RepetitionAcrossTwoConfigMap;repetition_across_two_config_map.yaml;my-config-map-1.yaml,my-config-map-2.yaml",
+    "RepetitionAcrossTwoConfigMapExplicitNamespace;repetition_across_two_config_map_explicit_namespace.yaml;my-config-map-1.yaml,my-config-map-2.yaml",
+    "RepetitionAcrossTwoSecret;repetition_across_two_secret.yaml;my-secret-1.yaml,my-secret-2.yaml",
+    "RepetitionInsideConfigMapAfter;repetition_inside_config_map_after.yaml;my-config-map.yaml",
+    "RepetitionInsideConfigMapBefore;repetition_inside_config_map_before.yaml;my-config-map.yaml",
+  })
+  void shouldCheckCrossFileNonCompliant(String root, String mainFile, String otherFiles) {
+    String folder = "DuplicatedEnvironmentVariables/CrossFileTests/NonCompliant/" + root + "/";
+    String[] otherFilesSplit = Arrays.stream(otherFiles.split(",")).map(file -> folder + file).toArray(String[]::new);
+    KubernetesVerifier.verify(folder + mainFile, check, otherFilesSplit);
+  }
+
+  @ParameterizedTest
+  @CsvSource(delimiter = ';', value = {
+    "InvalidConfigMap;invalid_config_map.yaml;",
+    "NoRepetitionMixEnvAndConfigMap;no_repetition_mix_env_and_config_map.yaml;my-config-map.yaml",
+    "NoRepetitionMixEnvAndSecret;no_repetition_mix_env_and_secret.yaml;my-secret.yaml",
+    "NoRepetitionWithTwoConfigMap;no_repetition_with_two_config_map.yaml;my-config-map-1.yaml,my-config-map-2.yaml",
+    "NoRepetitionWithTwoSecret;no_repetition_with_two_secret.yaml;my-secret-1.yaml,my-secret-2.yaml",
+    "RepetitionAcrossConfigMapAndSecretButDifferentNamespace;repetition_across_config_map_and_secret_but_different_namespace.yaml;my-config-map.yaml,my-secret.yaml",
+    "UnknownConfigMap;unknown_config_map.yaml;",
```

Comment:
Is it not better to use `Arguments.arguments()` instead of CSV? It is hard to read the above CSV.